core/include/detray/intersection/ray_line_intersector.hpp

```diff
@@ -0,0 +1,93 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include <type_traits>
+
+#include "detray/definitions/qualifiers.hpp"
+#include "detray/intersection/intersection.hpp"
+#include "detray/intersection/ray_plane_intersector.hpp"
+#include "detray/propagator/track.hpp"
+
+namespace detray {
+
+/** This is an intersector struct for line surface
+ * The intersectino point is obtained by calculating a transform3 object of
+ * virtual plane Virtual plane has a normal vector same with the direction of
+ * the track
+ */
+struct ray_line_intersector {
+
+    using intersection_type = line_plane_intersection;
+    using transform3 = __plugin::transform3<detray::scalar>;
+    using vector3 = __plugin::vector3<detray::scalar>;
+
+    /** Function to obtain the virtual plane whose normal vector is same with
+     * direction of the track
+     *
+     * @param trf the surface to be intersected
+     * @param track the track information
+     *
+     * @return the transform3 of the virtual plane
+     */
+    template <typename track_t>
+    transform3 line_to_planar_transform3(const transform3 &trf,
+                                         const track_t &track) const {
+
+        // z direction of virtual plane parallel to the track direction
+        const vector3 new_z = track.dir();
+
+        // y direction of virtual plane identical to line direction
+        const vector3 new_y = getter::vector<3>(trf.matrix(), 0, 2);
```

Comment:
Sorry, if I have to ask a stupid question, but are ```new_z``` and ```new_y``` orthogonal? They should be, if they span a coordinate system for a plane or not?

Reply:
Good catch - it is removed (See the description that I updated again...)